test/api/unit/authorizers/site.test.js

```diff
@@ -1,216 +1,272 @@
-const expect = require("chai").expect
-const factory = require("../../support/factory")
+const crypto = require('crypto');
+const { expect } = require('chai');
 const nock = require('nock');
 const sinon = require('sinon');
+const factory = require('../../support/factory');
 const githubAPINocks = require('../../support/githubAPINocks');
-const authorizer = require("../../../../api/authorizers/site.js")
+const authorizer = require('../../../../api/authorizers/site.js');
 const siteErrors = require('../../../../api/responses/siteErrors');
 const FederalistUsersHelper = require('../../../../api/services/FederalistUsersHelper');
 
-describe("Site authorizer", () => {
-  describe(".create(user, params)", () => {
-    it("should resolve", done => {
+describe('Site authorizer', () => {
+  describe('.create(user, params)', () => {
+    beforeEach(() => {
+      Promise.all([
+        factory.organization.truncate(),
+        factory.role.truncate(),
+      ]);
+    });
+
+    afterEach(() => {
+      Promise.all([
+        factory.organization.truncate(),
+        factory.role.truncate(),
+      ]);
+    });
+
+    it('should resolve', async () => {
+      const user = await factory.user();
+      const params = {
+        owner: crypto.randomBytes(3).toString('hex'),
+        repository: crypto.randomBytes(3).toString('hex'),
+        defaultBranch: 'main',
+        engine: 'jekyll',
+      };
+
+      const expected = await authorizer.create(user, params);
+      return expect(expected).to.be.undefined;
+    });
+
+    it('should resolve for user with organizations', async () => {
+      const [user, org, role] = await Promise.all([
+        factory.user(),
+        factory.organization.create(),
+        factory.role.create(),
+      ]);
       const params = {
-        owner: crypto.randomBytes(3).toString("hex"),
-        repository: crypto.randomBytes(3).toString("hex"),
-        defaultBranch: "main",
-        engine: "jekyll",
+        owner: crypto.randomBytes(3).toString('hex'),
+        repository: crypto.randomBytes(3).toString('hex'),
+        defaultBranch: 'main',
+        engine: 'jekyll',
+        organizationId: org.id,
+      };
+
+      await org.addUser(user, { through: { roleId: role.id } });
+      const expected = await authorizer.create(user, params);
+
+      return expect(expected).to.be.undefined;
+    });
+
+    // eslint-disable-next-line consistent-return
+    it('should throw an error for user with organizations and no organizationId specified', async () => {
+      const [user, org, role] = await Promise.all([
+        factory.user(),
+        factory.organization.create(),
+        factory.role.create(),
+      ]);
+      const params = {
+        owner: crypto.randomBytes(3).toString('hex'),
+        repository: crypto.randomBytes(3).toString('hex'),
+        defaultBranch: 'main',
+        engine: 'jekyll',
+      };
+
+      await org.addUser(user, { through: { roleId: role.id } });
+
+      try {
```

Comment:
When using try/catch as here, if this doesn't actually throw an error, the test will still pass. For this to work you would have to do something like:
```
let error;
try {
  await authorizer.create(user, params);
} catch (e) {
  error = e;
}
expect(error.message).to.equal(siteErrors.ORGANIZATION_REQUIRED);
```

The pattern we have been using for async errors is:
```
const error = await authorizer.create(user, params).catch(e => e);
expect(error).to.be.an('Error'); // probably not 100% necessary
expect(error.message).to.equal(siteErrors.ORGANIZATION_REQUIRED);
```
where the extra check for an 'Error' addresses the wacky case where it resolves to something with a matching`message` property, which seems highly unlikely

Reply:
I went down this route but since we are using `throw`, it doesn't return the error object.  To clean it up, I can use the `catch` appended to the async function like `await authorizer.create(...).catch(err => expect(err)...)`